### PR TITLE
JBEHAVE-1475 Perform beforStep(String) hook for comment, ignorable an…

### PR DESCRIPTION
**Pull request by uarlouski**
…d pending steps

### CODE REVIEW

**Review by valfirst — APPROVED** [review]

File: jbehave-core/src/main/java/org/jbehave/core/model/Step.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.jbehave.core.model;
+
+import org.jbehave.core.steps.StepCreator.StepKind;
+
+public class Step {
+
+    private final StepKind stepKind;
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/model/Step.java]: `StepType type`

File: jbehave-core/src/main/java/org/jbehave/core/steps/StepCreator.java [hunk 2]
```diff
@@ -918,58 +944,48 @@ public void annotatedOn(Method method) {
...
         }
+    }
 
-        @Override
-        public String asString(Keywords keywords) {
-            return stepAsString;
-        }
+    public static enum StepKind
+    {
+        PARAMETERISED,
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/steps/StepCreator.java]: EXECUTABLE

**Review by valfirst — CHANGES_REQUESTED** [review]

File: jbehave-core/src/main/java/org/jbehave/core/steps/StepCreator.java [hunk 3]
```diff
@@ -918,58 +944,48 @@ public void annotatedOn(Method method) {
...
+            return perform();
         }
+    }
 
-        @Override
-        public String asString(Keywords keywords) {
-            return stepAsString;
-        }
+    public static enum StepExecutionType
+    {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/steps/StepCreator.java]: use egyptian notation:
```
... {
...
}
```

**Review by valfirst — CHANGES_REQUESTED** [review]
Also please info to the docs about breaking changes

File: distribution/src/site/content/migration-paths.html [hunk 4]
```diff
@@ -37,6 +37,8 @@ <h2>From 4.x to 5.0</h2>
         Use storyTimeouts instead.</li>
     <li>Stop reporting BeforeStories/AfterStories steps as stories. New hooks are introduced: org.jbehave.core.reporters.StoryReporter#beforeStoriesSteps and org.jbehave.core.reporters.StoryReporter#afterStoriesSteps</li>
     <li>Fix typo in StoryReporter method: lifecyle -> lifecycle</li>
+    <li>Replace StoryReporter#beforeStep(String) hook with StoryReporter#beforeStep(Step)</li>
+    <li>Invoke StoryReporter#beforeStep(Step) hook before all steps (parmeterized, comment, pending, ignorable)</li>
```
Comment (valfirst) [inline distribution/src/site/content/migration-paths.html]:
```
... before all steps (including comment, pending, ignorable), previously the hook was invoked only for parameterized steps
```

**Review by valfirst — CHANGES_REQUESTED** [review]
`JBEHAVE-1475 Improve beforStep hook to be invoked for all steps` ?

**Review by valfirst — APPROVED** [review]

**Comment by uarlouski** [comment]
In master https://github.com/jbehave/jbehave-core/commit/9967402c52d7d8a70539f306feca592156015f9b

**Pull request closed without merging**